src/components/Table/GridTable.tsx

```diff
@@ -345,29 +331,44 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     const headerRows: RowTuple<R>[] = [];
     const filteredRows: RowTuple<R>[] = [];
 
+    // Misc state to track our nested card-ification, i.e. interleaved actual rows + chrome rows
+    const nestedCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style);
+
     // Depth-first to filter
     function visit(row: GridDataRow<R>): void {
-      if (row.kind === "header") {
-        headerRows.push([row, makeRowComponent(row)]);
-        return;
-      }
-
       const passesFilter =
         filters.length === 0 ||
         filters.every((filter) =>
           columns.map((c) => applyRowFn(c, row)).some((maybeContent) => matchesFilter(maybeContent, filter)),
         );
+      // Even if we don't pass the filter, one of our children might, so we continue on after this check
```

Comment:
I'm thinking that a child passing the filter means we should probably include it's parent, especially for nested cards? Or even just tables in general. It's not what we do now, but not sure why/if it's super on purpose.

Reply:
I agree. I think the parent should be shown when a child matches the filter regardless of whether it is the nested style. It would give better context to the user. 

FWIW: I think I like the term "matchesFitler" vs "passesFilter". When first reading your comment I thought "...pass the filter..." meaning passing it as a prop or something. Not a big deal, but just thinking.

Reply:
Yeah, will probably do the "show parent if child matches" is a separate/follow up breaking change.

Also agree on the variable name, will rename.